pkg/virt-controller/watch/migration.go

```diff
@@ -159,6 +159,15 @@ func (c *MigrationController) execute(key string) error {
 	migration := obj.(*virtv1.VirtualMachineInstanceMigration)
 	logger := log.Log.Object(migration)
 
+	// this must be first step in execution. Writting the object
+	// when api version changes ensures our api stored version is updated.
+	if !controller.ObservedLatestApiVersionAnnotation(migration) {
```

Comment:
What happens if a user fetches a VMI which was not yet processed and tries to update it with the old version and we didn't to a write?

Reply:
it's fine. it doesn't actually matter who does the write or with what version. We just need to ensure a write occurs after an api change occurs. That's the intent behind this code block.

The user can't impact the storage api version. If they update the vmi using v1alpha3 or using v1, it doesn't matter internally.  it is functionally the exact same.

Reply:
What happens if there was no write?

Reply:
It depends.

If the Storage version changed, then we aren't guaranteed the object is written internally to use the new Storage version.

If only a new API version is introduced, any attempt to PATCH the existing object using a client with the new API version will fail until the object is UPDATED using the previous client.

Reply:
This is really so ugly did we make sure that there is a bug report regarding to this?